server/utils/taskUtils.ts

```diff
@@ -1,9 +1,84 @@
 import { Task } from '../@types/shared'
+import { TableRow } from '../@types/ui'
+import { DateFormats } from './dateUtils'
+import { sentenceCase } from './utils'
 
-type GroupedTask = { allocated: Array<Task>; unallocated: Array<Task> }
+const DUE_DATE_APPROACHING_DAYS_WINDOW = 3
+
+const allocatedTableRows = (tasks: Array<Task>): Array<TableRow> => {
```

Comment:
I think it's fine to be repetitive here, rather than reusing the result of `unallocatedTableRows` as that `splice` there seems like a magic number and makes the code difficult to parse. I did wonder about doing the reverse, but removing items by an index seems equally magical 🪄 

Maybe we could have small functions for each row item, i.e:

```
const personRow = (task: Task) => {
  return {
    text: task?.person?.name || '',
  }
}

etc, etc
```

And then call them like so:

```
const allocatedTableRows = (tasks: Array<Task>): Array<TableRow> => {
    return tasks.map(t => [
      personRow(t),
     ....
    ])
}
```

Reply:
Yeah I was on the fence about whether to couple the two. I wanted to show that the `allocatedTableRows` is just the `unallocatedTableRows` with an person that it's allocated to and avoid having to duplicate all the logic for handling if an optional property isn't defined. 
I quite like your suggestion, will give it a go